test/org/sagebionetworks/bridge/play/controllers/ConsentControllerMockedTest.java

```diff
@@ -163,8 +167,14 @@ public void canWithdrawConsent() throws Exception {
         JsonNode node = BridgeObjectMapper.get().readTree(response);
         assertEquals("User has been withdrawn from the study.", node.get("message").asText());
         
+        // Should call the service and withdraw
         verify(consentService).withdrawConsent(any(Study.class), any(User.class), captor.capture(), any(Long.class));
         assertEquals("Because, reasons.", captor.getValue().getReason());
+        
+        // Should update the user's session to be non-consented
+        ArgumentCaptor<UserSession> sessionCaptor = ArgumentCaptor.forClass(UserSession.class);
+        verify(cacheProvider).setUserSession(sessionCaptor.capture());
+        assertFalse(sessionCaptor.getValue().getUser().doesConsent());
```

Comment:
This doesn't actually test that the user session was updated. The user session is a mock object with no override on doesConsent(), so it always returns false regardless.

A better thing to do is to create a stub User object that we can update and then either (a) move the user session management to ConsentController.withdrawConsent() so you can test it here or (b) move it to ConsentServiceImpl and move the test there as well.